CONTRIBUTING.md

```diff
@@ -3,39 +3,58 @@ CONTRIBUTING
 
 First of all: Thanks for your interest in contributing to WoltLab Suite Core! However, you have to meet some requirements in order to get your changes accepted.
 
-**Notice:** This is the development tree of WoltLab Suite Core representing the upcoming version of WoltLab Suite, if you wish to submit pull requests for WSC 3.0.x, please select the branch "3.0". 
+**Notice:** This is the development tree of WoltLab Suite Core representing the upcoming version of WoltLab Suite, if you wish to submit pull requests for stable versions, please select the branch you want to work on. 
 
 General requirements
 --------------------
 - **API changes are undesirable**, we want to maintain full backwards compatibility to WSC 3.0.x
 - Testing is the key, you MUST try out your changes before submitting pull requests. It saves us and yourself a lot of time.
 - The code SHOULD be written by yourself, otherwise you have to check the license beforehand with regard to compatibility and give the proper credit to the original author.
+- For getting into developing with WoltLab Suite Core you should take a look at the [WoltLab Suite Documentation](https://docs.woltlab.com/) for developers.
 
 Files
 -----
-- Unix newlines (\n) MUST be used in every file (php, tpl, less, js, etc.)
-- All files MUST be saved in ASCII encoding, except language files, which MUST be UTF-8 encoded
+- Unix newlines (\n) MUST be used in every file (php, tpl, scss, js, ts, etc.)
+- All `*.php`- and `*tpl`-files MUST be saved in ANSI/ASCII encoding, except language files, which MUST be UTF-8 encoded
+- `*.xml`-files MUST be encoded with UTF-8, but omit the BOM (byte-order-mark)
 
 Formatting
 ----------
-- Tabs MUST be used for indentation, you HAVE TO use a tab size of 8
-    - empty lines MUST be indentated as deep as the previous line
-- All identifiers and comments MUST be written in English
+- Files MUST be formatted according to the coding standard [PSR-12](https://www.php-fig.org/psr/psr-12/).
+- The best way is to use an IDE supporting EditorConfig for getting the correct settings automatically
 - PHP
     - The closing PHP tag MUST be omitted
     - Every file MUST end with a newline character (\n)
+- JavaScript
+    - Make sure every compiled file uses unix newlines (\n)
+- TypeScript
+    - You MUST intend lines with two (2) spaces; tab-characters are not allowed
```

Comment:
Actually you could just refer to Prettier, because this is what we use for TypeScript exclusively.